var/spack/repos/builtin/packages/py-macs2/package.py

```diff
@@ -13,13 +15,38 @@ class PyMacs2(PythonPackage):
     homepage = "https://github.com/taoliu/MACS"
     pypi = "MACS2/MACS2-2.2.4.tar.gz"
 
+    version("2.2.7.1", sha256="ad2ca69bdd02a8942a68aae23133289b5c16ba382bcbe20c39fabf3948929de5")
     version("2.2.4", sha256="b131aadc8f5fd94bec35308b821e1f7585def788d2e7c756fc8cac402ffee25b")
-
-    depends_on("python@3.5:", type=("build", "run"))
+    version("2.1.4", sha256="e4966d001914320829ab859c7bc8e92c6410aa7bdbddfd00b7625e9a0fb15c97")
+    version("2.1.3.3", sha256="00959e523f45ed92b8429f55944eca6984623ac008d7cdb488c3ffe59c21984a")
+    version(
+        "2.1.1.20160309", sha256="2008ba838f83f34f8e0fddefe2a3a0159f4a740707c68058f815b31ddad53d26"
+    )
+
+    # patch to correctly identify python-3.10 as greater than required version
+    patch(
+        "https://github.com/macs3-project/MACS/pull/497.patch?full_index=1",
+        sha256="eaff891b9b3c6a910bd5d454dcc6e21288c8d1ad4d6d6f77e370bc8f90921cbd",
+        when="^python@3.10:",
```

Comment:
Does this patch work for all versions? I would at least add `@:2.2.7.1` since it won't be needed in the future (if there ever is a new release someday).

Reply:
The 2.1 versions would not need it because they are python-2 only. I missed version 2.2.4 in there. Technically, it would need the patch, but I see that it does not apply cleanly to that version.

Reply:
Ah, version 2.2.4 does not build with python-3.10 anyway so a version constraint is in order.